### PR TITLE
Migrating to Typescript

### PR DESCRIPTION
**Pull request by seiyab**
## What this PR changes?
- Migrates Flow -> TypeScript
- Migrates [standard](https://standardjs.com/) -> [ts-standard](https://github.com/standard/ts-standard) (Official variant for TypeScript)
- Adds lint and type check for ui into build setting
- Resolves (or sometimes ignores) type errors.

## What I'm planning after this PR
- Opening issue about `any` or `ts-ignore` so that digdag-ui will become more reliable
- Opening issue about `eslint-disable` so that digdag-ui will become more relieable
- Submitting PRs that resolve some parts of issues above

## For reviewers, and for contributors who will come here from commit history
- I didn't resolve many typing / lint problems so that I can work this PR feasibly
    - So `any`, `ts-ignore` and `eslint-disable` are not desired ones. Resolving them should be welcome

### CODE REVIEW

File: digdag-ui/console.tsx [hunk 1]
```diff
@@ -45,90 +46,86 @@ import {
...
-    invalid: (key:string) => void;
-  }) => void;
-  scrub: Scrubber;
-}
-
-/* eslint-disable */
-// because of the global type, this results in an eslint error (when it's not)
-type ConsoleConfig = {
-  url: string;
+interface AuthItem {
```
Comment (seiyab) [inline digdag-ui/console.tsx:75]: ts-standard did `type` -> `interface`

File: digdag-ui/console.tsx [hunk 2]
```diff
@@ -45,90 +46,86 @@ import {
...
+  key: string
```
Comment (seiyab) [inline digdag-ui/console.tsx:76]: ts-standard removed trailing `,` `;` in type definitions

**Comment by seiyab** [comment]
@szyn 
Could you review this?
It is likely to conflict every digdag-ui changes.

**Review by szyn — APPROVED** [review]
Sorry for being late to review this PR, it looks good to me.

**Comment by seiyab** [comment]
I found that current `master` fails on type check (`./gradlew digdag-ui:checkUi`)
```
console.tsx(2041,34): error TS2769: No overload matches this call.
  Overload 1 of 2, '(data: Data, options: InflateFunctionOptions & { to: "string"; }): string', gave the following error.
    Argument of type 'Data | null' is not assignable to parameter of type 'Data'.
      Type 'null' is not assignable to type 'Data'. 
  Overload 2 of 2, '(data: Data, options?: InflateFunctionOptions | undefined): Uint8Array', gave the following error.
    Argument of type 'Data | null' is not assignable to parameter of type 'Data'.

> Task :digdag-ui:checkUi FAILED
```

@szyn @yoyama 
Should I add `./gradlew digdag-ui:checkUi` or `npm run type` into GitHub actions?
I'm confused by another error in `./gradlew check`.
I've supposed that `./gradlew check` should run on CI.
```
> Task :digdag-standards:test                                                                                                                                                                               
                                                                                                                                                                                                            
io.digdag.standards.operator.td.TdBaseTdJobOperatorTest > initializationError FAILED                                                                                                                        
    org.objenesis.ObjenesisException at SunReflectionFactoryHelper.java:54                                                                                                                                  
        Caused by: java.lang.reflect.InvocationTargetException at NativeMethodAccessorImpl.java:-2
            Caused by: java.lang.IllegalAccessError at ClassLoader.java:-2
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.eclipse.jetty.util.BufferUtil (file:~/.gradle/caches/modules-2/files-2.1/org.eclipse.jetty/jetty-util/9.3.11.v20160721/1812ffd5a04698051180d582c146ca80
7760c808/jetty-util-9.3.11.v20160721.jar) to field java.nio.MappedByteBuffer.fd
WARNING: Please consider reporting this to the maintainers of org.eclipse.jetty.util.BufferUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

304 tests completed, 1 failed                       

> Task :digdag-standards:test FAILED
```

**Pull request merged**